networkx/drawing/nx_pylab.py

```diff
@@ -51,6 +53,1091 @@
+        Weather to remove the ticks from the axes of the matplotlib object.
+
+    Raises
+    ------
+    NetworkXError
+        If a node or edge is missing a required parameter such as `pos` or
+        if `new_draw` receives an argument not listed above.
+
+    ValueError
+        If a node or edge has an invalid color format, i.e. not a color string,
+        rgb tuple or rgba tuple.
+
+    Returns
+    -------
+    The input graph. This is potentially useful for dispatching visualization
+    functions.
+    """
+    import warnings
+
+    warnings.warn(
+        "The new_draw function is temporary, designed to exposue the new visualization API\nbefore it replaces the current nx.draw implementation.",
+        category=DeprecationWarning,
+        stacklevel=3,
+    )
+
+    from collections import Counter
+
+    import matplotlib as mpl
+    import matplotlib.pyplot as plt
+    import numpy as np
+
+    defaults = {
+        "pos": "pos",
+        "node_visible": True,
+        "node_color": "#1f78b4",
+        "node_size": 300,
+        "node_label": {
+            "size": 12,
+            "color": "#000000",
+            "family": "sans-serif",
+            "weight": "normal",
+            "alpha": 1.0,
+            "h_align": "center",
+            "v_align": "center",
+            "bbox": None,
+        },
+        "node_shape": "o",
+        "node_alpha": 1.0,
+        "node_border_width": 1.0,
+        "node_border_color": "face",
+        "edge_visible": True,
+        "edge_width": 1.0,
+        "edge_color": "#000000",
+        "edge_label": {
+            "size": 12,
+            "color": "#000000",
+            "family": "sans-serif",
+            "weight": "normal",
+            "alpha": 1.0,
+            "bbox": {"boxstyle": "round", "ec": (1.0, 1.0, 1.0), "fc": (1.0, 1.0, 1.0)},
+            "h_align": "center",
+            "v_align": "center",
+            "pos": 0.5,
+            "rotate": True,
+        },
+        "edge_style": "-",
+        "edge_alpha": 1.0,
+        "arrowstyle": "-|>" if G.is_directed() else "-",
+        "arrowsize": 10 if G.is_directed() else 0,
+        "edge_curvature": "arc3",
+        "edge_source_margin": 0,
+        "edge_target_margin": 0,
+        "hide_ticks": True,
+    }
+
+    # Check arguments
+    for kwarg in kwargs:
+        if kwarg not in defaults:
+            raise nx.NetworkXError(
+                f"Unrecongized visualization keyword argument: {kwarg}"
+            )
+
+    if canvas is None:
+        canvas = plt.gca()
+
+    if kwargs.get("hide_ticks", defaults["hide_ticks"]):
+        canvas.tick_params(
+            axis="both",
+            which="both",
+            bottom=False,
+            left=False,
+            labelbottom=False,
+            labelleft=False,
+        )
+
+    ### Helper methods and classes
+
+    # I hope that the order-perserving nature of dicts in python 3.7+
+    # means that G.nodes() will always return the same order.
```

Comment:
Yes -- the order preserving nature of dict means G.nodes will always return the same order.... even in Pypy this is true. (they started the whole dict-order feature and CPython implemented it slightly later).
So I think this comment can be removed.

Reply:
Agreed, it's from an early stage of development.